### PR TITLE
IEP-1262 Changing the configuration should also change the mode

### PR DESCRIPTION
**Pull request by sigmaaa**
## Description

Now, when we change the configuration it automatically sets the mode to Run/Debug accordingly 

Fixes # ([IEP-1262](https://jira.espressif.com:8443/browse/IEP-1262))

## Type of change

Please delete options that are not relevant.
- New feature (non-breaking change which adds functionality)

## How has this been tested?
- Create a project -> change configuration to the debug configuration -> the mode should be Debug -> Select launch configuration -> the mode should be Run

**Test Configuration**:
* ESP-IDF Version:
* OS (Windows,Linux and macOS):

## Dependent components impacted by this PR:

- Launch Bar

## Checklist
- [x] PR Self Reviewed
- [ ] Applied Code formatting
- [ ] Added Documentation
- [ ] Added Unit Test
- [ ] Verified on all platforms - Windows,Linux and macOS

## Summary by CodeRabbit

- **New Features**
  - Enhanced handling of launch configurations and modes.
  - Improved resource deletion wizard with more robust method implementations.
  
- **Refactor**
  - Streamlined import statements and refactored methods for better readability and efficiency.
  
- **Bug Fixes**
  - Corrected logic in dialog handling to ensure accurate user input processing.
  
- **Style**
  - Added braces for code blocks and improved spacing for better code quality and readability.

### CODE REVIEW

**Review by alirana01 — APPROVED** [review]
LGTM!

**Comment by AndriiFilippov** [comment]
@sigmaaa hi ! 
Tested under: 
OS - Windows 10 

change the configuration it automatically sets the mode to Run/Debug accordingly ✔️ 
LGTM 👍

**Review by AndriiFilippov — APPROVED** [review]

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/LaunchBarListener.java [hunk 1]
```diff
@@ -49,6 +53,41 @@ public static void setIgnoreTargetChange(boolean status)
 		targetChangeIgnored = status;
 	}
 
+	@Override
+	public void activeLaunchDescriptorChanged(ILaunchDescriptor descriptor)
+	{
+		descriptor.getAdapter(ILaunchConfiguration.class);
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/LaunchBarListener.java:59]: Do you want to remove this as we have not using the returned adapter?

**Review by kolipakakondal — APPROVED** [review]
LGTM

**Comment by kolipakakondal** [comment]
**Follow-up behavior** -   Would you also like to handle the case where changing the mode (Run to Debug and vice versa) updates the launch configuration in the dropdown menu?  I understand we currently handle this during launch with a pop-up, but if there's only one configuration, we could simply switch it and eliminate the pop-up choice for the user.

Jul-04-2024 12-18-57

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/LaunchBarListener.java (hunk 1 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/LaunchBarListener.java]: done

**Comment by sigmaaa** [comment]
> **Follow-up behavior** - Would you also like to handle the case where changing the mode (Run to Debug and vice versa) updates the launch configuration in the dropdown menu? I understand we currently handle this during launch with a pop-up, but if there's only one configuration, we could simply switch it and eliminate the pop-up choice for the user.
> 
> Jul-04-2024 12-18-57 [ Jul-04-2024 12-18-57 ](https://private-user-images.githubusercontent.com/8463287/345725582-47e0d86c-aa14-4a70-915a-aabc6dc75642.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAxNjg2NjcsIm5iZiI6MTcyMDE2ODM2NywicGF0aCI6Ii84NDYzMjg3LzM0NTcyNTU4Mi00N2UwZDg2Yy1hYTE0LTRhNzAtOTE1YS1hYWJjNmRjNzU2NDIuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MDVUMDgzMjQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTg0MDFiZDJjMTc1YTNhNWJmOTFiMTIyMTQ4NzYzMjcwMGEzNjM0OTk5ZjUwMDFmZTRhODgwMzM2ODM0ZmNhNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.KUzsX01gP-2Nw5B-7ACh0lTqdPP5Xm0GVeTxAU8DtpI) [ ](https://private-user-images.githubusercontent.com/8463287/345725582-47e0d86c-aa14-4a70-915a-aabc6dc75642.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAxNjg2NjcsIm5iZiI6MTcyMDE2ODM2NywicGF0aCI6Ii84NDYzMjg3LzM0NTcyNTU4Mi00N2UwZDg2Yy1hYTE0LTRhNzAtOTE1YS1hYWJjNmRjNzU2NDIuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MDVUMDgzMjQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTg0MDFiZDJjMTc1YTNhNWJmOTFiMTIyMTQ4NzYzMjcwMGEzNjM0OTk5ZjUwMDFmZTRhODgwMzM2ODM0ZmNhNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.KUzsX01gP-2Nw5B-7ACh0lTqdPP5Xm0GVeTxAU8DtpI)

Hi @kolipakakondal,

thanks for the review. Implemented in the latest PR

**Comment by kolipakakondal** [comment]
> thanks for the review. Implemented in the latest PR

Hi @sigmaaa I could still see the same issue where changing the mode is not changing the configuration. Can you please check.
Jul-10-2024 08-28-59


@AndriiFilippov fyi.

**Comment by sigmaaa** [comment]
> > thanks for the review. Implemented in the latest PR
> 
> Hi @sigmaaa I could still see the same issue where changing the mode is not changing the configuration. Can you please check. Jul-10-2024 08-28-59 [ Jul-10-2024 08-28-59 ](https://private-user-images.githubusercontent.com/8463287/347242779-48f94061-27b8-44f8-bd36-15dac7940ee0.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA2MTA2MzIsIm5iZiI6MTcyMDYxMDMzMiwicGF0aCI6Ii84NDYzMjg3LzM0NzI0Mjc3OS00OGY5NDA2MS0yN2I4LTQ0ZjgtYmQzNi0xNWRhYzc5NDBlZTAuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MTBUMTExODUyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2RjZTIwNjk4OGEyYTBiZWY1MmNlMzdlMDMxMTc5ZmM1MTAwYWY5MmQyMDkzNjU1NzQ4YmU4NGJkYTllZjM0MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.tpz4SadekpqcPsVfLLv0wnqAo74HY2wx44Tq0jQpmG0) [ ](https://private-user-images.githubusercontent.com/8463287/347242779-48f94061-27b8-44f8-bd36-15dac7940ee0.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA2MTA2MzIsIm5iZiI6MTcyMDYxMDMzMiwicGF0aCI6Ii84NDYzMjg3LzM0NzI0Mjc3OS00OGY5NDA2MS0yN2I4LTQ0ZjgtYmQzNi0xNWRhYzc5NDBlZTAuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MTBUMTExODUyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2RjZTIwNjk4OGEyYTBiZWY1MmNlMzdlMDMxMTc5ZmM1MTAwYWY5MmQyMDkzNjU1NzQ4YmU4NGJkYTllZjM0MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.tpz4SadekpqcPsVfLLv0wnqAo74HY2wx44Tq0jQpmG0)
> 
> @AndriiFilippov fyi.


Hi @kolipakakondal, this is the expected behavior for this PR since the goal of this PR was to change the mode based on the configuration, not vice versa. To change the configuration, click run or debug with the wrong configuration. 
Changing the configuration based on the mode automatically causes bugs and not consistent behavior, because changing the configuration causes changing the mode, so if we change the configuration based on the mode it could cause an infinite loop. Removing listeners didn't help to remove bugs completely. Check this PR where I tried to implement it https://github.com/espressif/idf-eclipse-plugin/pull/806. We can revisit this PR and try to implement it again, however, we considered the logic in this PR as a good alternative.
From the user's perspective, the current behavior looks like this:
The user has a launch configuration selected and he wants to debug the project, so he selects mode to the debug and sees the button "Debug" -> At this point he can click it and the configuration will be changed to the debug, so the debug action will be performed

**Comment by kolipakakondal** [comment]
Hi @sigmaaa I'm happy if you would like to revist the existing PR on this problem and fix this issue.

**Pull request merged**